05/app.js

```diff
@@ -0,0 +1,45 @@
+function Student(firstName, lastName) {
+	this.firstName = firstName;
+	this.lastName = lastName;
+	this.grades = {};
+}
+
+Student.prototype.addGrade = function (subject, grade) {
+	if (typeof this.grades[subject] === 'undefined') {
+		this.grades[subject] = [];
+	}
+	this.grades[subject].push(grade);
+};
+
+Student.prototype.getAverageGrade = function (subject) {
+	if (typeof subject === 'undefined') {
+		return (
+			[]
+				.concat(...Object.values(this.grades))
```

Comment:
Dwa razy wykonujesz tą samą operację tj. `...Object.values(this.grades)` - wartą jej wynik zapisać do zmiennej i wykorzystać ponownie ;)